nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -97,6 +97,7 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                     }
                     WorkStateHelper.setState(work.getId(), Work.State.RUNNING, stateTTL);
                 }
+                work.setWorkInstanceState(Work.State.RUNNING);
```

Comment:
Could you add a comment saying that this has semantics related to dead-letter-queue management?